src/main/scala/no/ndla/draftapi/service/WriteService.scala

```diff
@@ -338,12 +339,57 @@ trait WriteService {
             editorLabels = Seq.empty,
             created = new Date(0),
             updated = new Date(0),
-            updatedBy = ""
+            updatedBy = "",
+            availability = Availability.everyone,
+            grepCodes = Seq.empty,
+            copyright = article.copyright.map(e => e.copy(license = None)),
+            metaDescription = Seq.empty,
+            relatedContent = Seq.empty,
+            tags = Seq.empty
         )
 
       withComparableValues(changedArticle) != withComparableValues(existingArticle)
     }
 
+    private def shouldPartialPublish(existingArticle: Option[domain.Article],
+                                     changedArticle: domain.Article): Boolean = {
+      val isPublished = changedArticle.status.current == ArticleStatus.PUBLISHED || changedArticle.status.other
+        .contains(ArticleStatus.PUBLISHED)
+
+      val hasChangedPartialPublishField = existingArticle.forall(e => {
+        e.availability != changedArticle.availability ||
+        e.grepCodes != changedArticle.grepCodes ||
+        e.copyright.flatMap(e => e.license) != changedArticle.copyright.flatMap(e => e.license) ||
+        e.metaDescription != changedArticle.metaDescription ||
+        e.relatedContent != changedArticle.relatedContent ||
+        e.tags != changedArticle.tags
+      })
+
+      isPublished && hasChangedPartialPublishField
+    }
+
+    private def partialPublishIfNeeded(existingArticle: Option[domain.Article],
+                                       changedArticle: domain.Article,
+                                       language: String,
+                                       user: UserInfo): Try[domain.Article] = {
+      if (!shouldPartialPublish(existingArticle, changedArticle)) {
+        Success(changedArticle)
+      } else {
+        changedArticle.id match {
+          case None =>
+            Failure(ArticleVersioningException("Article supplied to partialPublish did not have an id. This is a bug."))
+          case Some(id) =>
+            partialPublish(id, PartialArticleFields.values.toSeq, language)
+            val newEditorNotes =
+              changedArticle.notes :+ domain.EditorNote("Artikkelen har blitt delpublisert",
```

Comment:
Skal vi vise kva felt som er delpublisert, eller blir det berre støy for brukeren?

Reply:
Da må jeg holde state på hva som har endret seg osv. blir en del omskriving isåfall, og da må vi printe det som et slags liste med elementer siden det kan være flere ting som har blitt endret, kan hende det blir litt støyete. Jeg hadde kanskje latt det stå og se om NDLA har et behov for den type informasjon?